src/PilatusCamera.cpp

```diff
@@ -296,6 +298,41 @@ void Camera::_resync()
     send("version");
 }
 
+//-----------------------------------------------------
+//
+//-----------------------------------------------------
+void Camera::_checkcmd()
```

Comment:
The available commands were already checked in Camera::_run, so why you need this extra function?

Reply:
Commands availability has to be checked when the camera is constructed.
Otherwise other objects may get wrong capabilities.
I use the first communications with the hardware to do the check, this is in connect method which has a lock to avoid conflicts with Camera::_run.
I guess that once it is checked on camera construction it is not needed on Camera::_run method.

In my case, 1M pilatus3 does not have setroi command. If this is discovered after creating the Interface Lima thinks that hardware roi exists and crashes.